roles/docker/tasks/install.yml

```diff
@@ -6,6 +6,13 @@
   when: inventory_hostname in groups['gnbsim_nodes']
   become: true
 
+- name: remove previous installations
+  shell:
+    cmd: |
+      for pkg in docker.io docker-doc docker-compose docker-compose-v2 podman-docker containerd runc; do sudo apt-get remove $pkg; done
+  when: inventory_hostname in groups['gnbsim_nodes']
+  become: true
```

Comment:
Why do you need to remove previous installation? Why not just `skipping` the installation-related tasks if Docker is already installed? That is, if the machine has already the Docker repository added to its `apt`, then skip adding the key and repository again and just proceed to install the required Docker-related packages, no?
Moreover, I think the issue is not about having Docker components already installed in the machine but, the issue I see is because there is conflict between the way the Docker repo is added to `apt` from Docker instructions (step 1 in https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository) and the way how Ansible adds the same repository

Reply:
I referred to the official Docker documentation, which advises uninstalling any conflicting packages before installing Docker Engine. However, I found that leaving those packages installed also worked fine, so I removed it from pull request.